fem/bilininteg.hpp

```diff
@@ -2536,11 +2544,50 @@ class VectorDiffusionIntegrator : public BilinearFormIntegrator
 
 private:
    DenseMatrix dshape, dshapedxt, pelmat;
-   DenseMatrix Jinv, gshape;
+   int vdim = -1;
+   DenseMatrix mcoeff;
+   Vector vcoeff;
 
 public:
-   VectorDiffusionIntegrator() { Q = NULL; }
-   VectorDiffusionIntegrator(Coefficient &q) { Q = &q; }
+   VectorDiffusionIntegrator() {}
+
+   /** \brief Integrator with unit coefficient for caller-specified vector dimension.
+
+      If the vector demension does not match the true dimension of the \c
+      FiniteElementSpace , the resulting element matrix will be mathematically
+      invalid. */
+   VectorDiffusionIntegrator(int vector_dimension) : vdim{vector_dimension} {}
+
+   VectorDiffusionIntegrator(Coefficient &q) : Q{&q} {}
+
+   /** \brief Integrator with scalar coefficient for caller-specified vector dimension.
+
+      The element matrix is block-diagonal with \c vdim copies of the element
+      matrix integrated with the \c Coefficient.
+
+      If the vector demension does not match the true dimension of the \c
+      FiniteElementSpace , the resulting element matrix will be mathematically
+      invalid. */
+   VectorDiffusionIntegrator(Coefficient &q, int vector_dimension) : Q{&q}, vdim{vector_dimension} {}
```

Comment:
Matter of taste, but in MFEM I guess we prefer using parenthesis for the constructors

```
: Q(&q), vdim(vector_dimension) {}
```